src/Shared/MSBuildNameIgnoreCaseComparer.cs

```diff
@@ -136,7 +125,8 @@ public static bool Equals(string compareToString, string constrainedString, int
             }
 
 #if RETAIL
-            if ((s_runningProcessorArchitecture != NativeMethodsShared.PROCESSOR_ARCHITECTURE_IA64) && (s_runningProcessorArchitecture != NativeMethodsShared.PROCESSOR_ARCHITECTURE_ARM))
+            if ((s_runningProcessorArchitecture != NativeMethodsShared.ProcessorArchitectures.Architecture_IA64)
+                && (s_runningProcessorArchitecture != NativeMethodsShared.ProcessorArchitectures.Architecture_ARM))
             {
                 // The use of unsafe here is quite a bit faster than the regular
```

Comment:
I wonder if this is true any more.  We should test it extensively at some point when we have a good perf infrastructure.